v2/apiserver/internal/api/mongodb/projects_store.go

```diff
@@ -205,16 +205,16 @@ function() {
 			{Key: "id", Value: 1},
 		},
 	)
-	cur, err := p.collection.Find(
-		ctx,
-		bson.M{
-			"spec.eventSubscriptions": bson.M{
-				"$elemMatch": preliminaryMatchCriteria,
-			},
-			"$where": where,
+	query := bson.M{
+		"spec.eventSubscriptions": bson.M{
+			"$elemMatch": preliminaryMatchCriteria,
 		},
-		findOptions,
-	)
+		"$where": where,
+	}
+	if event.ProjectID != "" {
+		query["id"] = event.ProjectID
+	}
+	cur, err := p.collection.Find(ctx, query, findOptions)
```

Comment:
This improvement, _besides_ being an overall improvement, eliminates the need to implement a `Contains()` member function on the generic `meta.List` type.